### PR TITLE
Store usage log on GitHub when S3 is not available

### PR DESCRIPTION
**Pull request by huydhn**
It turns out that we haven't uploaded the usage log to GitHub when S3 is not available (macos, rocm), for example, https://github.com/pytorch/pytorch/actions/runs/3325822440#artifacts only includes test-report, test-json, sccache stats, and build artifacts.

### CODE REVIEW

File: .github/actions/upload-test-artifacts/action.yml [hunk 1]
```diff
@@ -129,3 +129,13 @@ runs:
...
         path: test/**/*.xml
+
+    - name: Store Usage Logs on Github
+      uses: actions/upload-artifact@v3
+      if: inputs.use-gha
+      with:
+        # Add the run attempt, see [Artifact run attempt]
+        name: usage-log-runattempt${{ github.run_attempt }}-${{ inputs.file-suffix }}.zip
+        retention-days: 14
+        if-no-files-found: error
```
Comment (clee2000) [inline .github/actions/upload-test-artifacts/action.yml:140]: can this be changed to ignore or warn?

**Review by clee2000 — APPROVED** [review]

File: .github/actions/upload-test-artifacts/action.yml (hunk 1 above)
Comment (huydhn) [reply .github/actions/upload-test-artifacts/action.yml:140]: Yeah, let's stick with a warning here https://github.com/actions/upload-artifact/blob/main/action.yml#L16. I kind of copy this line from above

**Comment by huydhn** [comment]
Confirm that the usage logs are now available on GHA https://github.com/pytorch/pytorch/actions/runs/3342375847

**Pull request closed without merging**